lambdas/services/bulk_upload_metadata_preprocessor_service.py

```diff
@@ -0,0 +1,404 @@
+
+        return metadata_row
+
+    def update_record_filename(self, original_row: dict, updated_row: dict):
+        stripped_file_path = original_row.get("FILEPATH").lstrip("/")
+        original_file_key = self.practice_directory + "/" + stripped_file_path
+        new_file_key = updated_row.get("FILEPATH").lstrip("/")
+
+        logger.info(f"Renaming file `{original_file_key}` to `{new_file_key}`")
+        if original_file_key != new_file_key:
+            try:
+                self.s3_service.client.copy_object(
+                    Bucket=self.staging_store_bucket,
+                    CopySource={
+                        "Bucket": self.staging_store_bucket,
+                        "Key": original_file_key,
+                    },
+                    Key=new_file_key,
+                    MetadataDirective="COPY",
+                    TaggingDirective="COPY",
+                )
+            except ClientError as e:
+                error_code = e.response["Error"]["Code"]
+                if error_code == "NoSuchKey":
+                    logger.info(f"File {original_file_key} doesn't exist")
+                    return None
+                else:
+                    logger.error(
+                        f"Failed update filename for `{original_file_key}`: {e}"
+                    )
+                    return None
+
+            self.s3_service.client.delete_object(
+                Bucket=self.staging_store_bucket, Key=original_file_key
+            )
+
+        return updated_row
+
+    def move_original_metadata_file(self, file_key: str):
+        destination_key = f"{self.practice_directory}/{self.processed_folder_name}/{self.processed_date}/{METADATA_FILENAME}"
+        logger.info(
+            f"Moving original metadata file from {file_key} to {destination_key}"
+        )
+        try:
+            self.s3_service.copy_across_bucket(
+                self.staging_store_bucket,
+                file_key,
+                self.staging_store_bucket,
+                destination_key,
+            )
+            return True
+        except ClientError as e:
+            logger.error(
+                f"Failed to move metadata file '{file_key}' to '{destination_key}': {e}"
+            )
+            return False
+
+    @staticmethod
+    def extract_person_name_from_bulk_upload_file_name(
+        file_path: str,
+    ) -> tuple[str, str]:
+        document_number_expression = r".*?([\p{L}][^\d]*[\p{L}])(.*)"
+        expression_result = regex.search(
+            rf"{document_number_expression}", file_path, regex.IGNORECASE
+        )
+
+        if expression_result is None:
+            logger.info("Failed to find the person name in the file name")
+            raise InvalidFileNameException("Invalid patient name")
+
+        name = expression_result.group(1)
+        current_file_path = expression_result.group(2)
+
+        return name, current_file_path
```

Comment:
`person` and `patient` are being used interchangeably, I would suggest the following refactor:
```suggestion
    def extract_patient_name_from_bulk_upload_file_name(
        file_path: str,
    ) -> tuple[str, str]:
        document_number_expression = r".*?([\p{L}][^\d]*[\p{L}])(.*)"
        expression_result = regex.search(
            rf"{document_number_expression}", file_path, regex.IGNORECASE
        )

        if expression_result is None:
            logger.info("Failed to find the patient name in the file name")
            raise InvalidFileNameException("Invalid patient name")

        patient_name = expression_result.group(1)
        current_file_path = expression_result.group(2)

        return patient_name, current_file_path
```